src/db/write.ts

```diff
@@ -298,11 +298,13 @@ export class Write {
         }
       }
       const valueHash = await index.flush(this._dagWrite);
-      const {meta} = index;
-      meta.valueHash = valueHash;
```

Comment:
This was causing a lot of debugging trouble. I changed things to `readonly` and it got flagged and it fixed the bug.

Conclusion. We should use readonly more aggresively.